c/driver/postgresql/copy/writer.h

```diff
@@ -452,13 +454,21 @@ class PostgresCopyListFieldWriter : public PostgresCopyFieldWriter {
     constexpr int32_t ndim = 1;
     constexpr int32_t has_null_flags = 0;
 
-    const int32_t start = ArrowArrayViewListChildOffset(array_view_, index);
-    const int32_t end = ArrowArrayViewListChildOffset(array_view_, index + 1);
+    // TODO: the LARGE_LIST should use 64 bit indexes
```

Comment:
Just discovered this as an issue in the current implementation - if there happens to be a LARGE_LIST that has a gap between the start/end offsets exceeding 32 bits of storage then we have to raise, as it would exceed what postgres allows for the `dim` field

I figured easier done in a follow up, but happy to do in this as well

Reply:
I'm OK with a followup since I'm just going to merge this :)